articles/tokens/id_token.md

```diff
@@ -1,130 +1,104 @@
+The ID token, usually referred to as `id_token` in code samples, is a [JSON Web Token (JWT)](/jwt) that contains user profile data. It is consumed by the client and used to get user information like the user's name, email, and so forth, typically used for UI display. It was added to the OIDC specification as an optimization so the client can know the identity of the user, without having to make an additional network requests.
 
-## How to get the `id_token`
+The `id_token` conforms to an industry standard (IETF [RFC 7519](https://tools.ietf.org/html/rfc7519)) and contains three parts: a header, a body and a signature.
+- The header contains the type of token and the hash algorithm used on the contents of the token.  
+- The body, also called the payload, contains identity claims about a user.  There are some claims with registered names, for things like the issuer of the token, the subject of the token (who the claims are about), and the time of issuance.  Any number of additional claims with other names can be added, though care must be taken to keep the JWT within the browser size limitations for URLs.  
+- The signature which is used by the recipient of a JWT to validate the integrity of the information conveyed in the JWT.
 
-The `id_token` is returned when calling any of the Auth0 functions which invoke authentication.  This includes calls to the Lock widget, to the auth0.js library, or the libraries for other languages.  In our code samples for Lock and auth0.js for example, there is a variable called `id_token` in the callback function passed to Lock and auth0.js which receives the `id_token`.
+## How to get the ID token
 
-### Lock libraries:
+The `id_token` can be returned when calling any of the Auth0 functions which invoke authentication.  This includes calls to the Lock widget, to the auth0.js library, or the libraries for other languages. You can view the implementation details for retrieving the `id_token` at the [Lock web library](/libraries/lock) and [Auth0.js library](/libraries/auth0js) documents.
 
-* [Lock web library](/libraries/lock)
-* [Lock iOS library](/libraries/lock-ios)
-* [Lock Android library](/libraries/lock-android)
+## How to control the contents of the ID token
 
-### Auth0.js
+In order to retrieve an `id_token` the `responseType` should be set to `id_token`, both for client-side and server-side authentication flows. The attributes included in the issued `id_token` are controlled by the use of a [parameter called `scope`](/scopes).
```

Comment:
Non-pipeline2 customers should use `code` for server side flows and `token` for client side flows. Consider adding a warning panel.